fn.go

```diff
@@ -105,6 +105,14 @@ func (f *Function) RunFunction(_ context.Context, req *fnv1beta1.RunFunctionRequ
 			response.Fatal(rsp, errors.Wrap(err, "cannot decode manifest"))
 			return rsp, nil
 		}
+		// When decoding YAML into an Unstructured object, unquoted values like booleans or integers
+		// can inadvertently be set as annotations, leading to unexpected behavior in later processing
+		// steps that assume string-only values, such as GetAnnotations.
+		if _, _, err := unstructured.NestedStringMap(u.Object, "metadata", "annotations"); err != nil {
+			response.Fatal(rsp, errors.Wrap(err, "invalid function input: invalid %v"))
```

Comment:
Looks like you either wanted to use WrapF and pass the invalid annotation and value, or you left a dangling %v. Either way, I think it would be better to have a more explicit message here, e.g. `invalid annotations`, mentioning the resource gvk and either the resource name set via the given annotation or the metadata.name, if any of them is available.

Reply:
Example of the error message when rendering the 'inline' example with an invalid value of True set would now be:

`crossplane: error: cannot render composite resource: pipeline step "render-templates" returned a fatal result: invalid annotations in resource "iam.aws.upbound.io/v1beta1, Kind=User": .metadata.annotations accessor error: contains non-string key in the map: true is of the type bool, expected string`

It might be a little harder to include the resource name thats set via annotation since the function used to retrieve it (GetAnnotations) is the point of failure. Do you think this message is explicit enough, or should I also include the metadata.name if it exists?

Reply:
Looking good now, thanks 🙏